client/js/Core/componentsandbox.js

```diff
@@ -247,330 +200,11 @@ SCWeb.core.ComponentSandbox.prototype.createViewersForScStructs = function (cont
-            sceneElementTypes = sceneElementTypes.filter(type => type.isEdge());
-            if (filterList) triples = triples.filter(triple => !filterList.some(element => element === triple.get("_scene_edge").value));
-            triples = splitArray(triples, self.maxSCgTriplesNumber);
-            sceneElementTypes = splitArray(sceneElementTypes, self.maxSCgTriplesNumber);
-            return [triples, sceneElementTypes];
-        };
-
-        let scTemplate = new sc.ScTemplate();
-        scTemplate.triple(
-            sceneAddr,
-            [sc.ScType.EdgeAccessVarPosPerm, "_edge_from_scene"],
-            [sc.ScType.Unknown, "_scene_edge"],
-        );
-        let triples = (await scClient.templateSearch(scTemplate)).map((triple) => {
-            return {
-                connectorFromScene: triple.get("_edge_from_scene"),
-                sceneElement: triple.get("_scene_edge"),
-            };
-        });
-        let sceneElementTypes = await scClient.checkElements(triples.map(triple => triple.sceneElement));
-        [triples, sceneElementTypes] = filterTriples(triples, sceneElementTypes, null);
-
-        for (let i = 0; i < triples.length; ++i) {
-            const triple = triples[i];
-            triple.isAdded = true;
-            triple.sceneElementType = sceneElementTypes[i];
-
-            self.eventStructUpdate(triple);
-        }
-    };
-
-    const updateLinkWindow = async (linkAddr) => {
-        const forceAppendData = async (oldBucket) => {
-            for (let content of oldBucket) {
-                await self.onDataAppend(content.data);
-            }
-        };
-
-        const sliceAndForceAppendData = async () => {
-            const oldBucket = self.contentBucket.slice();
-            self.contentBucket = [];
-            await forceAppendData(await scClient.getLinkContents(oldBucket));
-        };
-
-        const appendData = async (element) => {
-            self.contentBucket.push(element);
-            if (self.appendContentTimeoutId) clearTimeout(self.appendContentTimeoutId);
-
-            if (self.contentBucket.length > self.contentBucketSize) {
-                clearTimeout(self.appendContentTimeoutId);
-                await sliceAndForceAppendData();
-            }
-            else {
-                self.appendContentTimeoutId = setTimeout(sliceAndForceAppendData, self.appendContentTimeout);
-            }
-        };
-
-        await appendData(linkAddr);
-    };
-
-    if (this.is_struct && this.eventStructUpdate) {
-        if (SCWeb.core.Main.viewMode === SCgViewMode.DistanceBasedSCgView) {
-            updateDistanceBasedSCgWindow(sceneAddr).then(result => {
-                if (!result) updateDefaultSCgWindow(sceneAddr).then(null);
-            });
-        } else {
-            updateDefaultSCgWindow(sceneAddr).then(null);
-        }
+SCWeb.core.ComponentSandbox.prototype.updateContent = async function (keyElement) {
```

Comment:
sc-web-core.js:2779 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading '0')
    at SCWeb.core.ComponentSandbox.updateContent (sc-web-core.js:2779:23)
getting this when starting websocket script